### DISCUSSION TITLE
Warning pop to make scams a little harder

### DISCUSSION THREAD

**Discussion by Darin755** [rustdesk/rustdesk#3988] [Ideas]
Is there any plans to add warning popup that warns the user that it could be a scam? It seems like this software could easily be abused for malicious intentions. It would be nice to have a simple warning popup that has a brief explanation on the danger of blindly accepting a connection.

**Comment by GameParrot** [comment]
Maybe add a warning that it could be a scam if someone from India is trying to connect (because most scammers are in India).

**Comment by Darin755** [comment]
> Maybe add a warning that it could be a scam if someone from India is trying to connect (because most scammers are in India).

Isn't that racist?

**Reply by petesimon** [reply]
no. only factual

**Comment by bmcgonag** [comment]
I understand the desire to warn someone about potential scams, but any software could be abused in just such a way.  I think a well worded warning might be useful a very small percentage of the time, but generally if someone is being scammed, they have been contacted by phone or some other means, and asked to install the application, and thus the scammer would simply tell them there is a warning, and they should ignore it unless anyone else tries to connect.

**Comment by GameParrot** [comment]
Another idea could be to always show the IP address of the remote user when they connect. This would make scammers not want to use the software.

**Comment by Darin755** [comment]
I was thinking a popup that says "You are about to give someone access to your computer. Proceed?". It probably should be worded to be a little more intimidating but you get the picture.

**Comment by GameParrot** [comment]
Another idea is it could say something like, "WARNING: YOU MAY BE BEING SCAMMED!
If you are on the phone with someone who asked you to install RustDesk, do not install and hang up immediately. They are likely a scammer trying to steal your money or other private information." during installation. The installer could also make you check a box saying that your were not asked to install RustDesk by someone you don't know on the phone.

**Reply by kolAflash** [reply]
You could even go a step further and make the user type _"I GIVE SOMEONE ELSE FULL ACCESS TO MY COMPUTER"_ into a text field.  
But the checkbox may be sufficient too.

**Reply by dinger1986** [reply]
> You could even go a step further and make the user type _"I GIVE SOMEONE ELSE FULL ACCESS TO MY COMPUTER"_ into a text field. But the checkbox may be sufficient too.

no chance, absolutely not! We arent trying to cripple RustDesk! We are trying to put in reasonable levels of warnings.

The reality is whatever we do the bad actors will still be able to use it! 

All they need to do is compile it without warnings, find and replace any mention of RustDesk, host it on a cloned website of RustDesk called something like MustDesk and get people to go there, no warnings, no hassle.

**Reply by rustdesk** [reply]
Yes，absolutely right. They can also use our old version.

**Comment by Darin755** [comment]
> Another idea is it could say something like, "WARNING: YOU MAY BE BEING SCAMMED! If you are on the phone with someone who asked you to install RustDesk, do not install and hang up immediately. They are likely a scammer trying to steal your money or other private information." during installation. The installer could also make you check a box saying that your were not asked to install RustDesk by someone you don't know on the phone.

I personally like your idea of a checkbox. It makes them slow down and read

**Comment by dzatoah** [comment]
Maybe we could say something like: "A legitimate remote supporter would never ask for login credentials, especially not  bank accounts, online payment systems or giftcards. Never share your login credentials with anyone you don't trust 100% and know personally! If you are in doubt please ask your local Administrator before proceeding."

**Reply by kolAflash** [reply]
An attacker does not need to ask for such data when he has remote access.

So I prefer the text + checkbox proposed by @GameParrot  
https://github.com/rustdesk/rustdesk/discussions/3988#discussioncomment-5564880

**Comment by rustdesk** [comment]
👍 
@21pages Let's think about where to put such message.

**Comment by dzatoah** [comment]
I would suggest to put this here:
![image](https://user-images.githubusercontent.com/16420080/222706938-b3f6855d-1726-43ed-8448-d73e6d4057c2.png)


Or maybe a dialog popup if you click on the accept button.

**Comment by rustdesk** [comment]
> on the accept button.

And with a checkbox `don't show it any more for dainel's requrest` or `don't show it any more for all requests`?

**Comment by dzatoah** [comment]
Mh, checkboxes are not that user friendly. I suggest that the OK button of the dialog says something like: `Yes, I trust <NAME>` or something like that. And then just don't show the dialog for 'daniel' in this case anymore.

**Reply by kolAflash** [reply]
Like that idea.

But skip the `<NAME>`!  
I am not sure what `<NAME>` is in RustDesk. But I guess it is a string which can be controlled by the attacker!?

So an attacker could put any string into there. Even something completly fake like:
`your grandfather. ... CHECK DONE: Idenitity verified as "your grandfather"`

**Comment by rustdesk** [comment]
@dinger1986 @grummbeer do you have any idea about this?

pop up a dialog when first time open RustDesk, or for each connection?

**Comment by dinger1986** [comment]
Yeah I think a pop up.

Also on the website.

I'd use a warning along these lines

```Something like Warning! Remote control software like RustDesk is commonly used by scammers, make sure whoever you are giving your is trustworthy```

**Reply by pixelstuff** [reply]
An educational message like this would be a lot better than making a dogmatic statement like: "If you are on the phone with someone who asked you to install RustDesk, do not install and hang up immediately."

**Comment by rustdesk** [comment]
> website

https://github.com/rustdesk/rustdesk/discussions/3669

**Comment by grummbeer** [comment]
Important.

Some thoughts

- Messages along the way the scammer leads the victim. Starts with the download/website.
- The danger depends on "Permissions" settings. 
  - Use "screen share" as default
  - place a message there, above the checkboxes if other than "screen share" is selected
- Destructive actions need a extra confirmation. Granting access can also be destructive. So show an confirmation dialog on every/each "Accept"
   - Add checkox to this dialog "trust this [ID]" 
      - Dialog is not shown on next "accept" connect from this ID
      - name fakeable?
      - IDs could exists twice? 
   - The "trust" must be revokable, in settings/somewhere at peers
- React to settings
   - Show permissions in confirmations dialog
   - If is only "screen shareing" > [visual indicator] "it is safe to accept", true?

**Reply by rustdesk** [reply]
> * Use "screen share" as default

This bring big inconvience to normal users.



> * Destructive actions need a extra confirmation. Granting access can also be destructive. So show an confirmation dialog on every/each "Accept"

Can you implement this?

**Reply by grummbeer** [reply]
Will give it try.

**Reply by grummbeer** [reply]
Is it true that the cm is not visible when client run in debug mode?

> [2023-06-08T08:32:48Z ERROR librustdesk::server::connection] ipc to connection manager exit: Failed to connect to connection manager

Is the id always unique or do we need a hash/uuid for server+id?

**Comment by dinger1986** [comment]
Remembering as well it's not really rustdesks responsibility or fault that scammers use it. Can't make it too difficult for people to use it or mess up the ui with loads of warnings. It'll scare off others from using it

**Comment by grummbeer** [comment]
True. 

It's about finding the right balance. Having it but not being annoying or mess up the UI.

We're not talking about nerds oder admins. I don't know exactly, but I think for a few years now most users are "common" users in home offices who click as soon as a "click there" reaches their ear.

**Comment by dinger1986** [comment]
The biggest risk isn't users in a business it's home users. The granny and grandads who don't know any better.

Some warnings will help for sure.

Even on the download screen and a count down so there's a delay and time for them to think?

**Comment by grummbeer** [comment]
> countdown

After the countdown the download starts automatically :-)

Edit: argh … yes … time to cancel

**Comment by dinger1986** [comment]
The issue isn't the download though. So a countdown with that warning on the screen would be fine.

The reality is if there's enough warnings bad actors just simply won't use it for risk of being found out!

**Comment by grummbeer** [comment]
> The reality is if there's enough warnings bad actors just simply won't use it for risk of being found out!

Not that bad, i think.

Who are the audience for such kind of messages?

- People who do not know at all what a "remote desktop" is or how it could harm them.
- Not people who are already experienced in getting help via remote desktop or who themselves control their workstations in the company via home office.

Right?

**Comment by dinger1986** [comment]
Yeah the audience is people who are likely to be scammed not professionals

**Comment by grummbeer** [comment]
> pop up a dialog when first time open RustDesk, or for each connection?

Would say website and on the first launch.

**Comment by dinger1986** [comment]
Yeah agreed!

**Comment by rustdesk** [comment]
👍 
> "WARNING: YOU MAY BE BEING SCAMMED! If you are on the phone with someone who asked you to install RustDesk, do not install and hang up immediately. They are likely a scammer trying to steal your money or other private information."

**Comment by dinger1986** [comment]
need to find the balance remember!

**Comment by Darin755** [comment]
I think it be a good idea to limit the number of connection attempts in a 5min period.

If the popup keeps coming up people may accept it just to make it go away

**Comment by dinger1986** [comment]
Yes limit the connections would be good as well.

The main thing is to make bad actors not use rustdesk. Unfortunately there's no way to stop them doing what they are doing!!

**Comment by dinger1986** [comment]
Yes limit the connections would be good as well.

The main thing is to make bad actors not use rustdesk. Unfortunately there's no way to stop them doing what they are doing!!

**Comment by test2a** [comment]
I have a proposition to make. There are dozens of anti scam YouTubers and on twitch. They are the real experts in both knowing what the target audience (older gen ornon technical people) language they might understand or how scammers work. 

My suggestion is, we, the community reach out to them and ask for their advice. They would surely be able to help in the ux part. 

Otherwise we are merely splitting hair discussing what should not what should not be because we simply don't know and are guessing.

**Comment by Darin755** [comment]
> I have a proposition to make. There are dozens of anti scam YouTubers and on twitch. They are the real experts in both knowing what the target audience (older gen ornon technical people) language they might understand or how scammers work.
> 
> My suggestion is, we, the community reach out to them and ask for their advice. They would surely be able to help in the ux part.
> 
> Otherwise we are merely splitting hair discussing what should not what should not be because we simply don't know and are guessing.

This is a really good idea. I would also try Reddit

**Comment by Darin755** [comment]
Thanks for converting this into a discussion. This is a much better place for this.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/4336#issuecomment-1555869754

**Comment by rustdesk** [comment]
How do you think about a warn banner on website like this.

![image](https://github.com/rustdesk/rustdesk/assets/71636191/bdcbeb1c-6888-44e0-855b-e3b6a0b6e450)

**Reply by dinger1986** [reply]
I think it shouldnt take away from the website, maybe it should pop up when you click download and and then a count down before downloading and then you have to click to download it?

**Reply by petesimon** [reply]
add the words "_received email, or a message on social media_" in there and it's great

**Reply by Prismaria** [reply]
"You may be being scammed" ? 

That sounds wrong, grammar-wise. Should've dropped the "being".

**Comment by syncbricks** [comment]
Simple - it should have better security than team viewer and any desk

**Comment by pixelstuff** [comment]
> "WARNING: YOU MAY BE BEING SCAMMED! If you are on the phone with someone who asked you to install RustDesk, do not install and hang up immediately. They are likely a scammer trying to steal your money or other private information."

I think the above message is a little too accusatory or dogmatic saying that all phone requests are automatically scams.

The quote below from @dinger1986 would have my vote, or some variation of it. A message that tries to educate the user to make a proper decision rather than a hard and fast rule which is not to be broken.

> Warning! Remote control software like RustDesk is commonly used by scammers, make sure whoever you are giving remote access to is trustworthy

**Reply by Prismaria** [reply]
On the contrary, while using a level-headed response like the 2nd one may sound nicer to the ears, it's less salient in terms grabbing the attention of people. And most people's attention span are very selective.

**Reply by LeSnake04** [reply]
I think in this case being overly dramatic is much better than the warning being too weak. If the person on the phone is an actual family member/ caretaker they can easily explain in but it should make the person installing it question if they ARE REALLY SURE they trust the person on the other side. And for actual family tech support  its IMO better anyway to set up remote desktop when you/or someone else they know meet them in person to set up remote connections.

A bit more convincing work being necessary for legit tech support is a small price to pay for the millions of dollars  the warning stops from going to scammers

**Comment by LeSnake04** [comment]
I think a additional strong warning for completely new connections makes a lot of sense. I'm pretty sure there are cases where the family uses RustDesk for legit tech support but a scammer finds that out and takes advantage of the user  being used to accepting connections from rustdesk. A strongly worded warning that this is a new device and this might  be a scam would help a lot by throwing them off. That waring should be opt out though in cases where frequent new connections to the client are intended.  

Though allowing some restriction on starting connections with new peers in the first place (like forcing 2fa or alerting a contact) would help even more.
